gk/main.c

```diff
@@ -552,6 +552,72 @@ gk_setup_rss(struct gk_config *gk_conf)
 	return ret;
 }
 
+static void
+gk_arp_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
+{
+	unsigned int i;
+	struct gk_config *gk_conf = arg;
+	struct gk_lpm *ltbl = gk_conf->lpm_tbl;
+
+	for (i = 0; i < ltbl->num_tunnels; i++) {
+		if (ltbl->tunnels[i].flow.proto == ETHER_TYPE_IPv4
+				&& memcmp(&ltbl->tunnels[i].flow.f.v6.dst,
+				map->ip_be, sizeof(struct in_addr)) == 0) {
+			ether_addr_copy(&map->ha,
+				&ltbl->tunnels[i].nexthop_mac);
+			break;
+		}
+	}
+
+	*pcall_again = true;
+}
+
+static void
+gk_nd_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
+{
+	unsigned int i;
+	struct gk_config *gk_conf = arg;
+	struct gk_lpm *ltbl = gk_conf->lpm_tbl;
+
+	for (i = 0; i < ltbl->num_tunnels; i++) {
+		if (ltbl->tunnels[i].flow.proto == ETHER_TYPE_IPv6
```

Comment:
When there is an update in the LLS block, it needs to update the corresponding information in GK block. However, the LLS block doesn't know exactly which entry in the GK block has been updated, so I think we cannot use ``arg`` to hold ``ltbl->tunnels[i]`` directly, instead it needs to find the updated entry.